src/yaml/YamlSchema.ts

```diff
@@ -0,0 +1,152 @@
+  }
+
+  /**
+   * Returns the JSON Schema to use for the  given application yaml file URI.
+   *
+   * @param applicationYamlUri the application yaml file URI
+   */
+  public async getSchema(applicationYamlUri: string): Promise<string | undefined> {
+    const yamlSchema = this.cache.get(applicationYamlUri);
+    if (yamlSchema) {
+      return yamlSchema.jsonSchema;
+    }
+    if (!this._languageClient) {
+      return undefined;
+    }
+    const params = {
+      uri: applicationYamlUri,
+      scopes: [1, 2]
+    };
+    return this._languageClient.sendRequest('microprofile/jsonSchemaForProjectInfo', params)
+      .then((result: JsonSchemaForProjectInfo) => {
+        const jsonSchema = result.jsonSchema;
+        this.cache.set(applicationYamlUri, result);
+        return jsonSchema;
+      }, (err) => {
+        console.error("Error while consumming 'microprofile/jsonSchemaForProjectInfo' request: " + err.message);
+      });
+  }
+
+  /**
+   *  Evict the cache according to the classpath changed of the given project URIs
+   *
+   * @param event the properties change event
+   */
+  public evict(event: MicroProfilePropertiesChangeEvent) {
+    // 1. collect all application.yaml uri which belong to the project URIs
+    const uriToDelete: Array<string> = [];
+    for (const entry of this.cache.entries()) {
+      if (event.projectURIs.includes(entry[1].projectURI)) {
+        uriToDelete.push(entry[0]);
+      }
+    }
+    // 2. evict the cache
+    uriToDelete.forEach(uri => {
+      this.cache.delete(uri);
+    });
+  }
+}
+
+let yamlSchemaCache: YamlSchemaCache;
+
+export async function registerYamlSchemaSupport(): Promise<YamlSchemaCache> {
+  const yamlPlugin: any = await activateYamlExtension();
+  if (!yamlPlugin || !yamlPlugin.registerContributor) {
+    // activateYamlExtension has already alerted to users for errors.
+    return;
+  }
+  yamlSchemaCache = new YamlSchemaCache();
+  // register for microprofile schema provider
+  yamlPlugin.registerContributor(MICROPROFILE_SCHEMA, requestYamlSchemaUriCallback, requestYamlSchemaContentCallback);
+  return yamlSchemaCache;
+}
+
+// find redhat.vscode-yaml extension and try to activate it to get the yaml contributor
+async function activateYamlExtension(): Promise<{ registerContributor: YamlSchemaContributor } | undefined> {
+  const ext = vscode.extensions.getExtension(VSCODE_YAML_EXTENSION_ID);
+  if (!ext) {
+    vscode.window.showWarningMessage('Please install \'YAML Support by Red Hat\' via the Extensions pane.');
+    return undefined;
+  }
+  const yamlPlugin = await ext.activate();
+
+  if (!yamlPlugin || !yamlPlugin.registerContributor) {
+    vscode.window.showWarningMessage('The installed Red Hat YAML extension doesn\'t support Kubernetes Intellisense. Please upgrade \'YAML Support by Red Hat\' via the Extensions pane.');
+    return undefined;
+  }
+
+  if (ext.packageJSON.version && !semver.gte(ext.packageJSON.version, '0.0.15')) {
```

Comment:
@tsmaeder if Che doesn't have the proper vscode-yaml version, will users be able to install a more recent version?